### ISSUE TITLE
ANSI Tests fail on OSX

### ISSUE THREAD

**Issue by lastcoolnameleft** [Azure/simdem#27]
```$ # Stripping ANSI escape sequances

To make it easier to write scripts we don't want to include ANSI
escape sequences (such as colors and text deocration) in the results
section. SimDem automatically strips these when capturing the results.

$ echo -e "Normal \e[4mUnderlined\e[24m Normal"
Normal \e[4mUnderlined\e[24m Normal

--- 0.05497598648071289 seconds execution time ---

...

Expected results:

Normal Underlined Normal

Actual results:

Normal \e[4mUnderlined\e[24m Normal
```

I tried staring at this for a while and how to fix it, but was unable to derive a fix on my own.

Confirmed issue on iTerm + Terminal in OSX.

**Comment by lastcoolnameleft** [comment]
According to `git bisect` this is the first bad commit:
https://github.com/Azure/simdem/commit/5771c7141449cd841d309e5111af4ed470c635be

It's a pretty big commit to hard to tell exactly where thing went wrong.

**Comment by lastcoolnameleft** [comment]
I take that back.  I was getting a false negative due to a different test error.  This seems to be the culprit:
https://github.com/Azure/simdem/commit/a353e2af4e6e5fd2b990810226c7a942ecd8aaae

On the top, is a screenshot from running in OSX and on the bottom is running on Ubuntu.

Looking at the test, it doesn't appear to be filtering out the escape characters since the Ubuntu test has the characters underlined.  Is the test functioning properly?  

![screen shot 2017-08-16 at 9 06 22 pm](https://user-images.githubusercontent.com/686999/29392983-dd6f89a6-82c6-11e7-9208-aaed9b3a95fb.png)

**Comment by SorraTheOrc** [comment]
This is misleading and should be explained better in the text of the test. The results we get are returned to the console in their native form (i.e. with ANSI codes). This is so that the user sees the expected formatting.

However, when we test the codes are stripped. This is to make it easier to write the results and to ensure they look better when rendered as markdown.

It looks like the console in Mac is not parsing these codes in the return for some reason. Further, it looks Mac is not stripping correctly.

I don't have access to a Mac to test on, an I'm on a phone right now so pointing at the specific code is hard. I'll update this thread when I get time at a keyboard.

**Comment by lastcoolnameleft** [comment]
@rgardler See PR #28 for my proposed solution.

Attached screenshot shows that OSX does correctly underline the text; however, it was when I ran "man echo" on OSX and Ubuntu that I noticed the difference.  It's just when running via pexpect that something wonky must happen.  

This was the answer which enlightened me:  https://stackoverflow.com/a/43528813/8475874

I then ran "man echo" on each and noticed that there is no -e option on echo for OSX.  I tried printf and voila!  Worked on both.

![screen shot 2017-08-16 at 11 27 43 pm](https://user-images.githubusercontent.com/686999/29395993-885f91b8-82da-11e7-9016-19af160e4ca7.png)
